File: src/Rac.VOne.AccountTransfer/Export/MUFJFactorExporter.cs
using Rac.VOne.Web.Models;

namespace Rac.VOne.AccountTransfer.Export
{
    public class MUFJFactorExporter : ExporterBase
    {
        public override string GetHeaderRecord() => Join(
            DivisionHeader,
            91,
            0,
            PaymentAgency.ConsigneeCode,
            Company.Kana,
            DueAt.ToString("MMdd"),
            PaymentAgency.BankCode,
            PaymentAgency.BankName,
            PaymentAgency.BranchCode,
            PaymentAgency.BranchName,
            PaymentAgency.AccountTypeId,
            PaymentAgency.AccountNumber,
            Dummy());

        public override string GetDataRecord(AccountTransferDetail detail) => Join(
            DivisionData,
            detail.TransferBankCode,
            detail.TransferBankName,
            detail.TransferBranchCode,
            detail.TransferBranchName,
            Dummy(),
            detail.TransferAccountTypeId,
            detail.TransferAccountNumber,
            detail.TransferAccountName,
            decimal.Truncate(detail.BillingAmount),
            detail.TransferNewCode,
            detail.TransferCustomerCode,
            0,
            Dummy());

        public override string GetTrailerRecord() => Join(
            DivisionTrailer,
            TotalCount,
            decimal.Truncate(TotalAmount),
            0,
            0,
            0,
            0,
            Dummy());

        public override string GetEndRecord() => Join(
            DivisionEnd,
            Dummy());
    }
}

File: web/client/src/app/model/account-transfer-export/mufj-factor-exporter.ts
import { ExporterBase } from 'src/app/model/account-transfer-export/exporter-base'
import { AccountTransferDetail } from 'src/app/model/account-transfer-detail.model'

export class MUFJFactorExporter extends ExporterBase
{
    public GetHeaderRecord():string{
        return this.JoinData([
            this.DivisionHeader,
            91,
            0,
            this.PaymentAgency.consigneeCode,
            this.Company.kana,
            this.GetMMdd(this.DueAt),
            this.PaymentAgency.bankCode,
            this.PaymentAgency.bankName,
            this.PaymentAgency.branchCode,
            this.PaymentAgency.branchName,
            this.PaymentAgency.accountTypeId,
            this.PaymentAgency.accountNumber,
            this.Dummy()]);
    }

    public GetDataRecord(detail:AccountTransferDetail):string{
        return this.JoinData([
            this.DivisionData,
            detail.transferBankCode,
            detail.transferBankName,
            detail.transferBranchCode,
            detail.transferBranchName,
            this.Dummy(),
            detail.transferAccountTypeId,
            detail.transferAccountNumber,
            detail.transferAccountName,
            parseInt(detail.billingAmount.toString()),
            detail.transferNewCode,
            this.PaymentAgency.consigneeCode,
            detail.transferCustomerCode,
            0,
            this.Dummy()]);
    }

    public GetTrailerRecord():string{
        return this.JoinData([
            this.DivisionTrailer,
            this.TotalCount,
            parseInt(this.TotalAmount.toString()),
            0,
            0,
            0,
            0,
            this.Dummy()]);
    }

    public GetEndRecord():string{
        return this.JoinData([
            this.DivisionEnd,
            this.Dummy()]);
    }
}
